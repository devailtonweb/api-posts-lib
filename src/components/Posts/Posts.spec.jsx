import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
    posts: [
        {
            id: 1,
            title: 'title 1',
            body: 'body 1',
            cover: 'img/img1.png'
        },
        {
            id: 2,
            title: 'title 2',
            body: 'body 2',
            cover: 'img/img2.png'
        },
        {
            id: 3,
            title: 'title 3',
            body: 'body 3',
            cover: 'img/img3.png'
        }
    ]
};

describe('<Posts />', () => {
    it('should render posts', () => {
        render(<Posts />);

        //const { debug } = render(<Posts />);
        //debug();

        expect(screen.getAllByRole('heading', { name: /Nenhum post encontrado/i })).toHaveLength(1);
        expect(screen.getAllByRole('button', { name: /Load more posts/i })).toHaveLength(1);
        // expect(screen.getAllByText(/body/i)).toHaveLength(3);
    });

    it('should not render posts', () => {
        render(<Posts />);
        expect(screen.queryByRole('heading', { name: /title/i })).not.toBeInTheDocument();
        expect(screen.queryAllByRole('heading', { name: /title/i })).toHaveLength(0);
    });

    it('should match snapshot', () => {
        const { container } = render(<Posts {...props} />);
        const { firstChild } = container;
        expect(firstChild).toMatchSnapshot();
    });
});
