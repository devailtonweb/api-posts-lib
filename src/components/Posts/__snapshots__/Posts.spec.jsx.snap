// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<section
  class="container"
>
  <div>
    <div
      class="search-container"
    >
      <input
        class="text-input"
        placeholder="Type of search"
        type="search"
        value=""
      />
    </div>
  </div>
  <div
    class="posts"
  >
    <h3>
      Nenhum post encontrado :(
    </h3>
  </div>
  <div
    class="container-button"
  >
    <button
      class="button"
      disabled=""
    >
      Load more posts
    </button>
  </div>
</section>
`;
